src/utils/helpers.js

```diff
@@ -149,6 +149,31 @@ const utils = {
       setTimeout(() => controller.abort(), timeout);
     }
   },
+  // Modified version of the stringify implementation of https://github.com/sindresorhus/query-string
+  stringify: (obj, prefix, objType) => {
+    const pairs = [];
+
+    Object.keys(obj).forEach((key) => {
+      const value = obj[key];
+      const encodedKey = utils.encodeURIComponentRFC3986(key);
+
+      let pair;
+
+      if (Array.isArray(value)) {
+        pair = utils.stringify(value, prefix ? `${prefix}[${encodedKey}]` : encodedKey, 'array');
+      } else if (typeof value === 'object') {
+        pair = utils.stringify(value, prefix ? `${prefix}[${encodedKey}]` : encodedKey, 'object');
+      } else if (objType === 'object') {
+        pair = `${prefix ? `${prefix}[${encodedKey}]` : encodedKey}=${utils.encodeURIComponentRFC3986(value)}`;
+      } else {
+        pair = `${prefix || encodedKey}=${utils.encodeURIComponentRFC3986(value)}`;
+      }
```

Comment:
Should we add a case to set `pair = ''` if the value is `undefined` or `null`? 🤔 I don't think it'll ever really happen where we'll have an undefined or null value, but might be good to safeguard against it.

Might be worth adding a test around this new function as well. This is what I tested with:

`utils.stringify({ a: '1', b: ['1,2'], c: ['2', '3'], d: [true, false], e: { f: ['g', 'h'] }, i: undefined, j: null })`

Reply:
[4d29e01](https://github.com/Constructor-io/constructorio-client-javascript/pull/213/commits/4d29e0140712fa818eceec0b9863fcb04e13e80b)